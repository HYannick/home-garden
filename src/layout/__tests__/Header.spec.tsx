import React from 'react';
import 'jest-dom/extend-expect';
import { cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { renderWithTheme } from '../../theme-wrapper';
import HeaderView from '../header/Header.view';

afterEach(cleanup);
describe('Header', () => {
  const props = {
    t: jest.fn(),
    now: 'now',
    username: 'Alita',
    imgUrl: 'source.gunnm.jpg',
  };
  it('should render properly', () => {
    const { container } = renderWithTheme(
      <Router> <HeaderView {...props}/> </Router>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should render the right username', () => {
    const { getByText } = renderWithTheme(
      <Router> <HeaderView {...props}/></Router>,
    );
    expect(getByText(/^Alita/)).toHaveTextContent('Alita');
  });
});
