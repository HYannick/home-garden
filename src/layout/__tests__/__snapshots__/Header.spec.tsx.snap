// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
<div>
   
  <div
    class="css-1a9pc2l-Header"
  >
    <div
      class="css-b2f7vh-Header"
    >
      <div
        class="css-ilwsms"
        color="#333"
        opacity="0.7"
      />
      <div
        class="css-prjb17"
        placeholder="#EFFFE2"
      >
        <img
          alt="cover"
          class="css-10dr8mc"
        />
      </div>
    </div>
    <div
      class="css-gki4ed-Header"
    >
      <p
        class="css-1fhp3d6"
        color="white"
      >
         
        Fri Sep 06 2019
      </p>
      <p
        class="css-1nsadt8"
        color="white"
      >
        header.hi
        ,Â 
        <span
          class="css-atu78c"
          color="#B4ED86"
        >
          Alita
        </span>
      </p>
    </div>
    <a
      class="css-1w28esw"
      href="/profile"
    >
      <div
        class="css-prjb17"
        placeholder="#EFFFE2"
      >
        <img
          alt="username"
          class="css-10dr8mc"
        />
      </div>
    </a>
  </div>
   
</div>
`;
